protobufs/livekit_models.proto

```diff
@@ -302,6 +305,26 @@ message ChatMessage {
   bool generated = 6; // true if the chat message has been generated by an agent from a participant's audio transcription
 }
 
+message RpcRequest {
+  string id = 1;
+  string method = 2;
+  string payload = 3;
+  uint32 response_timeout_ms = 4;
+  uint32 version = 5;
```

Comment:
How is the version going to be used? Is it SDK level or application level?

If application level, feels like a version in response would be good.

Reply:
SDK level - just thinking ahead that we might change the protocol enough down the road (e.g. adding streaming) that you'd want an outdated client to reject new requests and/or an updated client to reject old ones. I figure we'll add `if version != 1 { error }` to the recipient side right now just in case? I'm not sure it's really needed but it might be good to have.  In that case we might have version 2 send both version 1 & 2 calls, and new recipients would only process the 1s. i'm not sure - probably needs more thought. But I'm anxious about making a mistake in the protocol that we only discover a little down the road and decide we need to update it in a way that's not backwards compatible.

Reply:
Got it. We have used a high level protocol version to check for compatibility, but a message level would be good too. I am guessing apps will encode their own versions in the payload if needed and handle backwards compatiblity.

But, the challenge in this case becomes how far do you go. If we are version 10 (hopefully never), would requester send all 10 versions?

With protobuf, new fields can be added and older clients would not be able to read them. Would the protocol change in a way where a v2 would look completely different or if you send the added fields in v2, v1 receiver would read it as v1 and send something back which is completely wrong?

Yeah, something to think about.
It is possible to an `optional` version field later too, but it becomes a bit awkward on receiver side. It would be something like
```
    if versionPresent {
          version = versionField
          if version != myVersion {
               return error
          }
    } else {
        return error
    }
```
So, that extra check to see if version is present or not would be needed if optional version is added later.

Reply:
keeping version in - hoping to not need it!